components/playlist/resources/reducers/playlist_reducer.ts

```diff
@@ -12,14 +12,27 @@ import * as PlaylistMojo from 'gen/brave/components/playlist/mojom/playlist.mojo
 const playlistReducer: Reducer<Playlist.State|undefined> =
     (state: Playlist.State|undefined, action) => {
       if (state === undefined) {
-        state = { lists: [] }
+        state = { lists: [], currentList: undefined }
       }
 
       // TODO(sko) Handle each action properly.
       switch (action.type) {
         case types.PLAYLIST_LOADED:
-          state = { ...state, lists: [...action.payload as PlaylistMojo.Playlist[]] }
+          const playlists = action.payload as PlaylistMojo.Playlist[]
+          let currentList
+          if (state.currentList) { currentList = playlists.find(list => list.id === state!.currentList!.id) }
+          if (!currentList) { currentList = playlists[0] }
+
+          if (playlists.length && !currentList) {
+            console.error('there\'s no selected playlist eventhough we have playlists')
```

Comment:
Optional nit:

I have no idea what our style guide says about quotes but I'd be tempted to use one of the other string quotes instead of escaping the apostrophe

```suggestion
            console.error("there's no selected playlist even though we have playlists")
```

or 

```suggestion
            console.error(`there's no selected playlist even though we have playlists`)
```

Always gets me when I come back to JavaScript after using a language which only has one string quote character for a while :laughing:

Reply:
Nit: Also, I think `eventhough` should have a space

Reply:
> Always gets me when I come back to JavaScript after using a language which only has one string quote character for a while 😆

Indeed 😹. Unfortunately, our eslint seems to complain about back quote when it doesn't have template parameter.

<img width="604" alt="image" src="https://user-images.githubusercontent.com/5474642/182057088-c841ce24-2a30-43f7-9bcf-f229b3bbfd7a.png">

Reply:
👍 whatever makes it happy then 😆